src/components/input/Input.tsx

```diff
@@ -154,29 +157,37 @@ export class Input extends React.Component<IInputProps, IInputState> {
             invalid: !this.state.valid && contains(validatedInputTypes, this.props.type),
         }, this.props.innerInputClasses);
 
-        return (
-            <div className={classes} onClick={(e: React.MouseEvent<HTMLElement>) => this.handleClick(e)}>
-                <input
-                    id={this.props.id}
-                    className={innerInputClasses}
-                    type={this.props.type}
-                    defaultValue={!isUndefined(this.props.value) ? this.props.value : this.props.defaultValue}
-                    ref={(innerInput: HTMLInputElement) => this.innerInput = innerInput}
-                    onBlur={() => this.handleBlur()}
-                    onChange={() => this.handleChange()}
-                    onKeyUp={(event: React.KeyboardEvent<HTMLInputElement>) => this.handleKeyUp(event)}
-                    placeholder={this.props.placeholder}
-                    checked={!!this.props.checked}
-                    disabled={!!this.props.disabled}
-                    name={this.props.name}
-                    required
-                    readOnly={!!this.props.readOnly}
-                    autoFocus={!!this.props.autoFocus}
-                    step={this.props.type === 'number' ? 'any' : null}
-                />
-                {this.getLabel()}
-                {this.props.children}
-            </div>
-        );
+        const inputElements = [
+            <input
+                id={this.props.id}
+                className={innerInputClasses}
+                type={this.props.type}
+                defaultValue={!isUndefined(this.props.value) ? this.props.value : this.props.defaultValue}
+                ref={(innerInput: HTMLInputElement) => this.innerInput = innerInput}
+                onBlur={() => this.handleBlur()}
+                onChange={() => this.handleChange()}
+                onKeyUp={(event: React.KeyboardEvent<HTMLInputElement>) => this.handleKeyUp(event)}
+                placeholder={this.props.placeholder}
+                checked={!!this.props.checked}
+                disabled={!!this.props.disabled}
+                name={this.props.name}
+                required
+                readOnly={!!this.props.readOnly}
+                autoFocus={!!this.props.autoFocus}
+                step={this.props.type === 'number' ? 'any' : null}

```

Comment:
You are going to need a key on the input, the label and children.

Reply:
since i am destructuring the array i don't think i need that unless i'm wrong